src/controllers/csi/gc/controller.go

```diff
@@ -152,7 +176,29 @@ func getAllPinnedVersionsForTenantUUID(ctx context.Context, apiReader client.Rea
 		if uuid != tenantUUID {
 			continue
 		}
-		pinnedImages[dynakube.CodeModulesVersion()] = true
+		codeModuleVersion := dynakube.CodeModulesVersion()
+		if codeModuleVersion != "" {
+			pinnedVersions[codeModuleVersion] = true
+		}
+	}
+	return pinnedVersions, nil
+}
+
+func getAllDynakubes(ctx context.Context, apiReader client.Reader, namespace string) (*dynatracev1beta1.DynaKubeList, error) {
```

Comment:
i might be wrong, but it feels like this function already exists.
maybe check if it does and refactor accordingly, or ignore this comment if it doesn't

Reply:
didn't find anything 😢